hazelcast/src/test/java/classloading/ThreadLocalLeakTestUtils.java

```diff
@@ -231,4 +236,27 @@ private static boolean loadedByThisOrChild(Object o, ClassLoader cl) {
         }
         return false;
     }
+
+    private static String describeThread(Thread thread) {
+        return format("[%s] of type [%s]", thread.getName(), getPrettyClassName(thread.getClass()));
+    }
+
+    // enhance message with information about known thread locals issues
+    private static String checkKnownIssues(String message, Thread thread, Object key) {
+        if (thread != null && key != null) {
+            String threadClassName = getPrettyClassName(thread.getClass());
+            String keyClassName = getPrettyClassName(key.getClass());
+            if (threadClassName.equals("com.hazelcast.internal.networking.nio.NioThread")
```

Comment:
Hazelcast is started with a `FilteringClassLoader` so an `instanceof` check will fail here.